### ISSUE TITLE
feature request: add prio parameter for filtering rules

### ISSUE THREAD

**Issue by adschm** [jech/babeld#74]
So far, filtering rules are evaluated based on their order in the configuration (or uci config for OpenWrt).

This is rather inconvenient when using scripts to alter these configurations (particularly for uci config in OpenWrt), as it may be hard to control the position of a specific entry without rewriting everything again in the new order. A common case is when one wants to add redistribute filters for local routes _before_ the "redistribute local deny" filter that needs to be last.

Since the config is only evaluated "at start" and not updated dynamically, it should be simple enough to provide an ordering of the filtering rules by a prio parameter when reading them (of course, with a default prio where the order will be used again as default).

This would be a great help.

**Comment by jech** [comment]
Agreed, that's a good idea.  And it would not be too difficult to use the priority even if rules are added at runtime.

**Comment by DanielG** [comment]
@adschm Have you considered using the `-c file` option to include multiple config files before/after the main one? I do that in my setup to make the config a little more composable.

One thing that we'd have to patch to make that really workable is the ability to ignore missing config files though.

**Comment by jech** [comment]
> One thing that we'd have to patch to make that really workable is the ability to ignore missing config files though.

```sh
fn="/dev/null"
[-f "$filename" ] && fn="$filename"
babeld -c "$fn"
```
